### ISSUE TITLE
TrackerBase.Key bug and questions.

### ISSUE THREAD

**Issue by Epicguru** [Vanilla-Expanded/VanillaAchievementsExpanded#1]
When updating my mod to use this latest version of VAE I've seen that `TrackerBase.Key` must now implement a setter.

There is an infinite recursion bug that will crash the game when attempting to either set the Key value in `GenericTracker`, or use the copy constructor in GenericTracker (which in turn calls the Key setter). This happens because the [setter calls itself](https://github.com/Vanilla-Expanded/VanillaAchievementsExpanded/blob/30357f3222d7104350a594b362c82045087f30e8/1.5/Source/AchievementsExpanded/AchievementsExpanded/Tracker/Base/GenericTracker.cs#L18) - it seems that it should be setting the value of the field `key` rather than the property `Key`.

As far as I can tell there does not seem to be a reason for the abstract property to need a setter. The only place the setter is being used is in GenericTracker, and it was bugged there anyway, and everywhere else the implementation [seems to be blank](https://github.com/Vanilla-Expanded/VanillaAchievementsExpanded/blob/30357f3222d7104350a594b362c82045087f30e8/1.5/Source/AchievementsExpanded/AchievementsExpanded/Tracker/Anomaly/EntitiesInHoldTracker.cs#L17). Following in that example I will also leave my custom trackers will blank setters, but I find it strange. Normally I wouldn't bring up minor code quality stuff like this in mod projects, but seeing that this is a framework for use by other developers it seems reasonable to question it.

Thanks.

**Comment by juanosarg** [comment]
Yeah, that was all leftovers from a thing Mlie and me tried to do, and I just forgot to remove that stuff. Just ignore it
